packages/backend/src/api/controllers/report/filter/filterReportsByProjects.ts

```diff
@@ -1,36 +1,31 @@
-import { EthereumAddress } from '@l2beat/common'
-
 import { ProjectInfo } from '../../../../model/ProjectInfo'
 import { ReportRecord } from '../../../../peripherals/database/ReportRepository'
 
 export function filterReportsByProjects(
   reports: ReportRecord[],
   projects: ProjectInfo[],
 ): ReportRecord[] {
-  const bridges = getBridges(projects)
-  return reports.filter((report) => {
-    const bridge = bridges.get(report.bridge)
-    if (!bridge || bridge.sinceBlock > report.blockNumber) {
+  const projectAssetMap = getProjectAssetMap(projects)
+  return reports.filter(({ projectId, asset }) => {
+    if (!projects.some((project) => project.projectId === projectId)) {
       return false
     }
-    const tokenSinceBlock = bridge.tokens.get(report.asset)
-    if (tokenSinceBlock === undefined || tokenSinceBlock > report.blockNumber) {
+    const assetIds = projectAssetMap.get(projectId)
+    if (!assetIds || !assetIds.some((id) => id === asset)) {
       return false
     }
     return true
   })
 }
 
-function getBridges(projects: ProjectInfo[]) {
+// change
+// now tokens are filtered based on project not bridge
+// todo discuss it
```

Comment:
I think that ultimately there should be no or minimal filtering done. The filtering should occur in ReportUpdater where non-conforming records are deleted from the database